setup.py

```diff
@@ -97,19 +97,24 @@ class Tester(Command):
     """
 
     description = 'test the code using pytest'
-    user_options: List[Tuple[str, str, str]] = []
+    user_options = [
+        ('type=', None, 'Type of test to run'),
+    ]
 
     def initialize_options(self):
         """Set default values for options that this command supports."""
-        pass
+        self.type = None
 
     def finalize_options(self):
         """Set final values for options that this command supports."""
-        pass
+        if self.type not in {None, 'directx', 'cuda', 'rocm'}:    # replace with your test types
+            raise Exception(f'Invalid type: {self.type}')
 
     def run(self):
         """Run pytest."""
-        errno = os.system('python3 -m pytest -v --cov=superbench --cov-report=xml --cov-report=term-missing tests/')
+        command = 'python3 -m pytest -v --cov=superbench --cov-report=xml --cov-report=term-missing tests/'
+        command += f' -k test_{self.type}' if self.type else ''
```

Comment:
why this is needed if you set env to skip test?

Reply:
because I need to skip other benchmarks unit test, they can not successfully run on win, and I can not set a skipunittest for all other benchmarks

Reply:
not only ansible, I have tried, there're too many issues, some are permission issues when create file, some are bin issues, etc.. I even can hardly check all issues in short time

Reply:
updated to use pytest command + '-k' in pipeline file directly